src/libstore/local-store.cc

```diff
@@ -1383,7 +1383,7 @@ bool LocalStore::verifyStore(bool checkContents, RepairFlag repair)
             checkInterrupt();
             auto name = link.path().filename();
             printMsg(lvlTalkative, "checking contents of '%s'", name);
-            PosixSourceAccessor accessor;
+            PosixSourceAccessor accessor(std::filesystem::current_path().root_path());
```

Comment:
Please wrap all std::filesystem calls that can exceptions that can throw exceptions in a try-catch so that we can throw a nix error instead. Otherwise it breaks our error messages as the error bubbles up without any error context getting added.

Reply:
Here it looks like only the `current_path` call can throw an exception, as defined by implementation. In practice Windows, Linux and macOS seem to always provide a value.